packages/sync-actions/src/product-types-actions.js

```diff
@@ -31,189 +23,245 @@ export function actionsMapBase(diff, previous, next, config = {}) {
   })
 }
 
-const attributeDefinitionsActionsList = [
-  { action: 'changeLabel', key: 'label' },
-  { action: 'setInputTip', key: 'inputTip' },
-  { actionKey: 'newValue', action: 'changeInputHint', key: 'inputHint' },
-  { action: 'changeIsSearchable', key: 'isSearchable' },
-  {
-    actionKey: 'newValue',
-    action: 'changeAttributeConstraint',
-    key: 'attributeConstraint',
-  },
-]
-
-const getIsAnAttributeDefinitionBaseFieldChange = diff =>
-  diff.label ||
-  diff.inputHint ||
-  diff.inputTip ||
-  diff.attributeConstraint ||
-  diff.isSearchable
-
-function actionsMapEnums(attributeType, attributeDiff, previous, next) {
-  const addEnumActionName =
-    attributeType === 'enum' ? 'addPlainEnumValue' : 'addLocalizedEnumValue'
-  const changeEnumOrderActionName =
-    attributeType === 'enum'
-      ? 'changePlainEnumValueOrder'
-      : 'changeLocalizedEnumValueOrder'
-  const changeEnumLabelActionName =
-    attributeType === 'enum'
-      ? 'changePlainEnumValueLabel'
-      : 'changeLocalizedEnumValueLabel'
-  const buildArrayActions = createBuildArrayActions('values', {
-    [ADD_ACTIONS]: newEnum => ({
-      attributeName: next.name,
-      action: addEnumActionName,
-      value: newEnum,
-    }),
-    [CHANGE_ACTIONS]: (oldEnum, newEnum) => {
-      const oldEnumInNext = next.values.find(
-        nextEnum => nextEnum.key === oldEnum.key
-      )
-
-      // These `changeActions` would impose a nested structure among
-      // the accumulated `updateActions` generated by `buildArrayActions`
-      // In the end; we have to flatten the structure before we pass it back
-      // to the client.
-      const changeActions = []
-      if (oldEnumInNext) {
-        if (!deepEqual(oldEnum.label, oldEnumInNext.label)) {
-          changeActions.push({
-            attributeName: next.name,
-            action: changeEnumLabelActionName,
-            newValue: newEnum,
-          })
-        } else {
-          changeActions.push({
-            attributeName: next.name,
-            action: changeEnumOrderActionName,
-            value: newEnum,
-          })
+// this is nearly similar to `buildBaseAttributesActions`, however with a significant difference.
+// `buildBasAttributesActions` generates update-actions with help of `diff`,
+// which is an object consisting of flags which indicates different operations.
+// `generateBaseFieldsUpdateActions` only generate based on `previous` and `next`.
+export const generateBaseFieldsUpdateActions = (
+  previous,
+  next,
+  actionDefinition
+) => {
+  const isEmpty = value =>
+    [isNil, nextValue => nextValue === ''].some(operator =>
```

Comment:
This is confusing...

```js
const isEmpty = value => isNil(value) || String(value).trim() === ''
```

wouldn't work since you are defining `isEmpty` as a _single_ function which is `value => isNil(value)`, where as an empty string is also an important operation to run on the value

Reply:
I tried it and it work as expected:
```js
> const isNil = value => value === null || value === undefined;
undefined
> const isEmpty = value => isNil(value) || value === '';
// const isEmpty = function (value) { return isNil(value) || value === '' }
undefined
> isEmpty(null)
true
> isEmpty(undefined)
true
> isEmpty('')
true
> isEmpty(1234)
false
> isEmpty("abcd")
false
```
For me it looks much simpler than looping through functions but it is your code so if you preffer your solution, go for it.